### PR TITLE
Conditional PSI check to reflect changes done in 5.13

### PR DESCRIPTION
**Pull request by rhellstrom**
cpu.pressure 'full' is undefined for system-wide checks since 5.13 but still reported with values set to 0 for backwards compatibility. Made changes to reflect this for system-wide checks so that the conditional comparison is not made against the 0 value and instead fall back to 'some'.

https://www.kernel.org/doc/html/latest/accounting/psi.html

### CODE REVIEW

File: src/shared/condition.c
Comment (YHNdnzj) [inline src/shared/condition.c]: Please check c->type instead of doing string comparison

Comment (rhellstrom) [reply src/shared/condition.c]: Thanks, issue addressed

File: src/shared/condition.c
Comment (YHNdnzj) [inline src/shared/condition.c]:
```suggestion
                system_wide = true;

                pressure_path = path_join("/proc/pressure", pressure_type);
                if (!pressure_path)
```

**Review by YHNdnzj — APPROVED** [review]
Superficially looks good, but I'm no PSI expert. I'd like to wait for one more reviewer before merge.

File: src/shared/condition.c
Comment (poettering) [reply src/shared/condition.c]: yes, please keep the allocation and its immediate error handling together. And assign the boolean either before both or after both, but not in between it

File: src/shared/condition.c
Comment (poettering) [inline src/shared/condition.c]: hmm, was the "full" line ever useful before kernel 5.13? If it always was a mistake and not useful, maybe we should never look into it for the system wide stuff?

Comment (rhellstrom) [reply src/shared/condition.c]: Looking into it, it appears to have been used and reported values prior to 5.13. But according to the [commit](https://github.com/torvalds/linux/commit/890d550d7dbac7a31ecaa78732aa22be282bb6b8) where it was removed, it never made much sense.

Comment (bluca) [reply src/shared/condition.c]: Then it's fine to never use it, no need for the fallback

File: src/shared/condition.c
Comment (poettering) [inline src/shared/condition.c]: we shouldn't even ask for this if we are looking at the per-system file. i.e. this call should already be conditionalized.

maybe instead of a "bool system_wide" just keep a "PressureType preferred_pressure_type". that one you either initialize to PRESSUE_TYPE_FULL or to PRESSURE_TYPE_SOME int he big if block above.

Then you just pass that here as 2nd arg, so that we ask for the right thing right away.

And then you conditionalize the 2nd call to read_resource_pressure() below not on the the system_wide bool but instead simply whether preferred_pressure_type is PRESSURE_TYPE_FULL currently.

Comment (rhellstrom) [reply src/shared/condition.c]: Thanks for the guidance, 

PressureType preferred_pressure_type is now set at the beginning to 'full'.

If the check is at system level and for cpu pressure then preferred_pressure_type is set to 'some'.

And the error handling for -ENODATA is simplified again because it should only ever occur if a kernel without cpu.pressure 'full' support for cgroups was used.

File: src/shared/condition.c
Comment (YHNdnzj) [inline src/shared/condition.c]: Please consitionalize this branch on _FULL, since there's no need to fall back if it's already _SOME

Comment (rhellstrom) [reply src/shared/condition.c]: Sure! For the sake of code clarity or am I missing something? I figured we would never be in such a situation if we've made it this far as 'some' will always return data

**Review by bluca — APPROVED** [review]

**Pull request merged**
